### PR TITLE
Dummy index locking mechanism

### PR DESCRIPTION
**Pull request by guykeller**
## Status 
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
To avoid the possibility of race conditions between several builds with premium packs, introduced a locking mechanism for the dummy index.

### CODE REVIEW

File: Tests/Marketplace/prepare_public_index_for_private_testing.py [hunk 1]
```diff
@@ -105,13 +106,43 @@ def option_handler():
...
+
+    with open('lock.txt', 'rb') as lock_file:
+        dummy_index_lock_blob.upload_from_file(lock_file)
+
+
+def acquire_dummy_index_lock(public_storage_bucket, dummy_index_lock_path):
+    total_seconds_waited = 0
+    while is_dummy_index_locked(public_storage_bucket, dummy_index_lock_path):
+        if total_seconds_waited >= MAX_SECONDS_TO_WAIT_FOR_LOCK:
+            exit(1)
```
Comment (yaakovi) [inline Tests/Marketplace/prepare_public_index_for_private_testing.py:134]: https://stackoverflow.com/questions/6501121/difference-between-exit-and-sys-exit-in-python

File: Tests/Marketplace/prepare_public_index_for_private_testing.py [hunk 2]
```diff
@@ -105,13 +106,43 @@ def option_handler():
...
+    with open('lock.txt', 'w') as lock_file:
+        lock_file.write('locked')
+
+    with open('lock.txt', 'rb') as lock_file:
+        dummy_index_lock_blob.upload_from_file(lock_file)
+
+
+def acquire_dummy_index_lock(public_storage_bucket, dummy_index_lock_path):
+    total_seconds_waited = 0
+    while is_dummy_index_locked(public_storage_bucket, dummy_index_lock_path):
```
Comment (yaakovi) [inline Tests/Marketplace/prepare_public_index_for_private_testing.py:132]: 10 minutes with no output will fail the build (add a print once a minute)

File: Tests/Marketplace/prepare_public_index_for_private_testing.py [hunk 3]
```diff
@@ -124,31 +155,37 @@ def main():
...
-    change_packs_price_to_zero(public_index_folder_path)
-
-    private_packs, private_index_path, private_index_blob = update_index_with_priced_packs(private_storage_bucket,
-                                                                                           extract_destination_path,
-                                                                                           public_index_folder_path,
-                                                                                           changed_pack, True)
-
-    upload_modified_index(public_index_folder_path, extract_public_index_path, dummy_index_blob, build_number,
-                          private_packs)
+    try:
```
Comment (yaakovi) [inline Tests/Marketplace/prepare_public_index_for_private_testing.py:170]: a nicer solution is a `with` statement.

Comment (guykeller) [reply Tests/Marketplace/prepare_public_index_for_private_testing.py]: Done

**Review by yaakovi — APPROVED** [review]

**Pull request merged**